### PR TITLE
Remove ToJSON and FromJSON superclass constriants from ResourcefulData

### PR DESCRIPTION
**Pull request by gwils**

### CODE REVIEW

File: json-api-lib.cabal [hunk 1]
```diff
@@ -12,6 +12,7 @@ category:            Network
 build-type:          Simple
 maintainer:          Shirren Premaratne
```
Comment (shirren) [inline json-api-lib.cabal:13]: @gwils would you like to add your name to the list of maintainers?

File: json-api-lib.cabal [hunk 2]
```diff
@@ -12,6 +12,7 @@ category:            Network
...
 tested-with:         GHC == 8.4.3
+extra-source-files:  CHANGELOG.md
```
Comment (shirren) [inline json-api-lib.cabal:15]: @gwils nice, good idea

File: src/Network/JSONApi/Resource.hs [hunk 3]
```diff
@@ -77,7 +77,7 @@ instance (NFData a) => NFData (Resource a)
 {- |
 A typeclass for decorating an entity with JSON API properties
 -}
-class (ToJSON a, FromJSON a) => ResourcefulEntity a where
+class ResourcefulEntity a where
```
Comment (shirren) [inline src/Network/JSONApi/Resource.hs:80]: @gwils can you explain why making this type class more general is a good idea? Can we not just put `ToJSON` to the list?

Comment (gwils) [reply src/Network/JSONApi/Resource.hs:80]: I think it's useful to not require constraints that we're not using. We can move these constraints closer to the use site.

In the same way that we're using only `ToJSON, ResourcefulEntity` in our project, I can imagine another project where someone is only using `FromJSON`, so I'd rather assume neither.

File: json-api-lib.cabal (hunk 1 above)
Comment (gwils) [reply json-api-lib.cabal]: Done, thanks

**Pull request merged**